malcolm/modules/pandablocks/controllers/pandamanagercontroller.py

```diff
@@ -267,6 +271,20 @@ def _handle_change(self, k, v, bus_changes, block_changes, bit_out_changes):
         if block_name == "*METADATA":
             if field_name.startswith("LABEL_"):
                 field_name, block_name = field_name.split("_", 1)
+            elif field_name == "LAYOUT" and v:
+                # Set the layout table first so that related fields are set after
+                # possible deletion
+                self._json_layout = json.loads(v)
```

Comment:
```suggestion
                self._json_layout = json.loads("".join(v))
```
v is a list but this expects a string.